driver-core/src/main/com/mongodb/connection/BaseCluster.java

```diff
@@ -68,12 +67,12 @@
     private volatile boolean isClosed;
     private volatile ClusterDescription description;
 
-    BaseCluster(final ClusterId clusterId, final ClusterSettings settings, final ClusterableServerFactory serverFactory) {
+    BaseCluster(final ClusterId clusterId, final ClusterSettings settings, final ClusterableServerFactory serverFactory,
+                final ClusterListener clusterListener) {
```

Comment:
Now there are two ClusterListener's passed to the constructor, one in clusterSettings, and one in clusterListener.  That invites confusion.  Can we instead just call EventListenerSettingsHelper#getClusterListener from this constructor rather than from the DefaultClusterFactory?

Seems like what we should do is have the DefaultClusterFactory merge the ClusterListener list in ClusterSettings with the clusterListener passed into any of its #create methods that take one, constructing a new ClusterSettings if necessary with the merged list.  Then BaseCluster can just take ClusterSettings and be done with it.   It's exceedingly unlikely that any applications are even calling and of those other #create methods (they could even be deprecated).